src/main/java/uk/ac/cam/cl/dtg/segue/dao/users/PgUsers.java

```diff
@@ -666,14 +669,36 @@ public void updateUserLastSeen(final RegisteredUser user, final Date date) throw
     }
 
     @Override
-    public void incrementSessionToken(RegisteredUser user) throws SegueDatabaseException {
+    public Integer regenerateSessionToken(RegisteredUser user) throws SegueDatabaseException {
+        Integer newSessionToken = generateRandomTokenInteger();
+        this.updateSessionToken(user, newSessionToken);
+        return newSessionToken;
+    }
+
+    private static Integer generateRandomTokenInteger() {
+        Random random = new SecureRandom();
+        int newValue = random.nextInt();
+        // -1 is reserved for 'no assigned token', used for when a user is logged out for example
+        if (newValue != -1) return newValue;
```

Comment:
This /could/ return -1. It's not hugely likely, but it could. I suppose it could also return 0, which we also don't want?

Reply:
Is there something I'm missing here? It should only return a value that passes the check of not being -1, right? Otherwise it recursively calls the function for a different value, which is also subject to the check of not equalling -1. Unless we hit the infinitesimally small probability of it getting -1 every time and recursing until it runs out of memory or something, I don't think we should have an issue here?
0 isn't actually a problem in and of itself. The problem before was that the initial session token for _every_ user was _always_ 0, and further it would remain 0 until they used the `logout everywhere` option instead of the standard `logout` option.